### PR TITLE
amxIdSystem: add new ID submodule

### PR DESCRIPTION
**Pull request by nickjacob**
## Type of change

- [x ] Other

## Description of change

Adding the `amxIdSystem` userId submodule. This provides an `amxId` user ID and eid.

contact email of the adapter’s maintainer: prebid@amxrtb.com
- [ x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

PR in docs repo: https://github.com/prebid/prebid.github.io/pull/3026

### CODE REVIEW

**Comment by smenzer** [comment]
@nickjacob can you please also add test to `userId_spec.js` - you can follow the same updates as other modules

**Comment by nickjacob** [comment]
@smenzer thanks for pointing that out -- looks like when 5.0 got merged into master it caused some weird issues in userId_spec.js -- I tried to resolve it in a way that made sense (specifically here -- in master: https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/userId_spec.js#L532)

**Comment by nickjacob** [comment]
thanks @patmmccann  -- added to submodules.json

File: modules/amxIdSystem.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                return;
+              } catch (e) {
+                logError(`${NAME} invalid response`, responseText);
+              }
+            }
+
+            done(null);
+          },
+        },
+        params
```
Comment (smenzer) [inline modules/amxIdSystem.js:143]: when i add run your code on page (I'd recommend adding it to `userId_example.html` under integration examples), i get an error with the response from the server. i believe `params` should be `JSON.stringify(parms)` here as it shows as `[object Object]` in the network call. either that, or the server is not responding properly.

**Review by smenzer — CHANGES_REQUESTED** [review]

**Comment by nickjacob** [comment]
Thanks @smenzer -- sorry about that! I added the integration example & fixed that request.

I also removed the "timeout" option

File: modules/amxIdSystem.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+### Storage Settings
+
+The following settings are available for the `storage` property in the `userSync.userIds[]` object:
+
+| Param under `storage` | Scope    | Type         | Description                                                                      | Example   |
+| --------------------- | -------- | ------------ | -------------------------------------------------------------------------------- | --------- |
+| name                  | Required | String       | Where the ID will be stored                                                      | `"amxId"` |
+| type                  | required | String       | This must be `"html5"`                                                           | `"html5"` |
+| expires               | Optional | Number <= 30 | number of days until the stored ID expires. **Must be less than or equal to 30** | `30`      |
```
Comment (smenzer) [inline modules/amxIdSystem.md:43]: this is not an optional parameter if you are using the built-in storage

File: integrationExamples/gpt/userId_example.html [hunk 3]
```diff
@@ -226,6 +227,13 @@
             {
               "name": "criteo"
             },
+            {
+              "name": "amxId",
+              "storage": {
+                "type": "html5",
+                "name": "amxId"
```
Comment (smenzer) [inline integrationExamples/gpt/userId_example.html:234]: you must pass in `expires`

**Review by smenzer — CHANGES_REQUESTED** [review]
just need to fix `expires` and then this looks good to me

**Review by smenzer — APPROVED** [review]
LGTM

**Pull request merged**